packages/commonwealth/client/scripts/models/Thread.ts

```diff
@@ -32,6 +42,17 @@ export type Link = {
   title?: string;
 };
 
+// a list of thread fields that are an array (only includes 1st level arrays)
+// IMPORTANT: this should always be updated when thread modal updates array fields or
+// api response changes
+export const THREAD_ARRAY_FIELDS = [
+  'collaborators',
+  'chainEntities',
+  'attachments',
+  'versionHistory',
+  'associatedReactions'
+]
```

Comment:
whenever we add/remove a new array field to threads object, we should update this. This array is used to correctly update cache in react query

Reply:
> It is not used anywhere? 

It is used in the `cache` file 

> What about polls, links, associatedReactions?

I added `polls` array with my changes (it was not added before) -- this step will help in polls migration to RQ, links was also not used, the `links` array is not used properly so I didn't add extra complexity here (again the plan here is to use it in future when we migrate some more code to RQ)


> maybe some more distinguished typescript check or dynamic const that created list of array-fields from the Thread instead of hardcoded list

you are correct here, but given the above exceptions I decided to go with this. But the proper way is as you suggested .

Reply:
> @mzparacha seems like it's not being referenced anywhere? can we remove this variable?

do you mean the polls variable?

Reply:
The list of array fields that Marcin suggested we remove/change @mzparacha

Reply:
Marcin suggested about `associatedReactions`, `links` and `polls`. I only added `polls`, `links` are used a little but not fully. I will go ahead with removing `polls`

Reply:
Removed

Reply:
@mzparacha I don't understand what the array of strings itself _does_

Reply:
Ok so turns out I ended up not using it in favor of [this](https://github.com/hicommonwealth/commonwealth/blob/98abe8f9902bbefd5d812ea080dd7a26b6ce3cf1/packages/commonwealth/client/scripts/state/api/threads/helpers/cache.ts#L236-L239). Removed the `THREAD_ARRAY_FIELDS `